src/content/developers/docs/smart-contracts/testing/index.md

```diff
@@ -1,37 +1,204 @@
+
+You can also commission a security audit to give users higher assurances of smart contract safety. Audits benefit from extensive analysis performed by cybersecurity professionals and detect potential vulnerabilities or bugs that could break the smart contract functionality. 
+
+### 2. Bug bounties {#bug-bounties}
+
+A bug bounty is a financial reward given to an individual who discovers a vulnerability or bug in a program's code and reports to developers. Bug bounties are similar to audits since it involves asking others to help find defects in smart contracts. The major difference is that bug bounty programs are open to the wider developer/hacker community. 
+
+Bug bounty programs often attract a broad class of ethical hackers and independent security professionals with unique skills and experience. This may be an advantage over smart contract audits that mainly rely on teams who may possess limited or narrow expertise.
+
+## Testing tools and libraries {#testing-tools-and-libraries}
+
+### Unit testing tools {#unit-testing-tools}
+
+**Solidity-Coverage** - _A Solidity code coverage tool useful for testing smart contracts._
 - [GitHub](https://github.com/sc-forks/solidity-coverage)
 
-**Whiteblock Genesis -** **_An end-to-end development sandbox and testing platform for blockchain._**
+**Waffle** - _A framework for advanced smart contract development and testing (based on ethers.js)_.
+- [Documentation](https://ethereum-waffle.readthedocs.io/en/latest/)
+- [GitHub](https://github.com/TrueFiEng/Waffle) 
+- [Website](https://getwaffle.io/)
+
+**Remix Tests** - _A tool for testing Solidity smart contracts. Works underneath Remix IDE "Solidity Unit Testing" plugin which is used to write and run test cases for a contract._
+- [Documentation](https://remix-ide.readthedocs.io/en/latest/unittesting.html) 
+- [GitHub](https://github.com/ethereum/remix-project/tree/master/libs/remix-tests)  
+
+**OpenZeppelin Test Helpers** - _Assertion library for Ethereum smart contract testing. Make sure your contracts behave as expected!_
+- [GitHub](https://github.com/OpenZeppelin/openzeppelin-test-helpers) 
+- [Documentation](https://docs.openzeppelin.com/test-helpers) 
+
+**Truffle smart contract test framework** - _Automated testing framework to make testing your contracts a breeze._
+- [Documentation](https://trufflesuite.com/docs/truffle/testing/testing-your-contracts/) 
+- [Website](https://trufflesuite.com/) 
+
+**Brownie unit testing framework** - _Brownie utilizes Pytest, a feature-rich test framework that lets you write small tests with minimal code, scales well for large projects, and is highly extendable._
+- [Documentation](https://eth-brownie.readthedocs.io/en/v1.0.0_a/tests.html) 
+- [GitHub](https://github.com/eth-brownie/brownie) 
+
```

Comment:
Add foundry tests https://github.com/foundry-rs/foundry

Reply:
> Add foundry tests https://github.com/foundry-rs/foundry

Added this now.